### PR TITLE
20201019 tentative changes

### PR DESCRIPTION
**Pull request by 0-8-15**
Please review - maybe it works for you too.

### CODE REVIEW

**Comment by 0-8-15** [comment]
Wait.

Just learned that this depends on a piece of code which I have here locally but not in the upstream: `android_getFilesDir`.

**Comment by 0-8-15** [comment]
In my code, `android_getFilesDir` is used in `libraries/lambdanative/system.c` around line 215 like this:

    sys_appdir=android_getFilesDir();

File: loaders/android/AndroidManifest.xml.in [hunk 1]
```diff
@@ -22,6 +22,6 @@
     @ANDROID_XML_SERVICES@
   </application>
 @ANDROID_XML_PERMISSIONS@
-<uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE"/>
+<uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE" android:required="true" />
```
Comment (mgorges) [inline loaders/android/AndroidManifest.xml.in:25]: android:required exists in [uses-feature](https://docs.huihoo.com/android/4.2///guide/topics/manifest/uses-feature-element.html) but not [uses-permission](https://developer.android.com/guide/topics/manifest/uses-permission-element)?

**Comment by mgorges** [comment]
> sys_appdir

Do you mean [libraries/liblambdanative/system.c#L155](https://github.com/part-cw/lambdanative/blob/master/libraries/liblambdanative/system.c#L155) as my file only has 203 lines?

File: loaders/android/bootstrap.java.in [hunk 2]
```diff
@@ -158,7 +158,7 @@ public class @SYS_APPNAME@ extends Activity implements @ANDROID_JAVA_IMPLEMENTS@
     // start EVENT_IDLE 
     if(idle_tmScheduleRate > 0) idle_tm.scheduleAtFixedRate(idle_task, 0, idle_tmScheduleRate);
 
-    nativeInstanceInit();
+    nativeInstanceInit(ln_getFilesDir());
```
Comment (mgorges) [inline loaders/android/bootstrap.java.in:161]: Sorry, but where is `ln_getFilesDir()` declared?

**Comment by 0-8-15** [comment]
OK, that was just a guess.  Forget it.

Am Mon, 19 Oct 2020 14:02:12 -0700
schrieb Matthias Görges <notifications@github.com>:

> @mgorges commented on this pull request.
> 
> 
> 
> > @@ -22,6 +22,6 @@  
>      @ANDROID_XML_SERVICES@
>    </application>
>  @ANDROID_XML_PERMISSIONS@
> -<uses-permission
> android:name="android.permission.WRITE_EXTERNAL_STORAGE"/>
> +<uses-permission
> android:name="android.permission.WRITE_EXTERNAL_STORAGE"
> android:required="true" />
> 
> android:required exists in
> [uses-feature](https://docs.huihoo.com/android/4.2///guide/topics/manifest/uses-feature-element.html)
> but not
> [uses-permission](https://developer.android.com/guide/topics/manifest/uses-permission-element)?

**Comment by 0-8-15** [comment]
Am Mon, 19 Oct 2020 14:11:57 -0700
schrieb Matthias Görges <notifications@github.com>:

> > sys_appdir  
> 
> Do you mean
> [libraries/liblambdanative/system.c#L155](https://github.com/part-cw/lambdanative/blob/master/libraries/liblambdanative/system.c#L155)
> as my file only has 203 lines?

Precisely.

File: loaders/android/bootstrap.java.in (hunk 2 above)
Comment (0-8-15) [reply loaders/android/bootstrap.java.in:161]: ah, the other half missing.  :-/

in [bootstrap.java.in](https://github.com/part-cw/lambdanative/blob/master/loaders/android/bootstrap.java.in#L90) line 90ff. I have

      public String ln_getFilesDir() {
          return getFilesDir().toString();
     }

**Comment by mgorges** [comment]
Can we merge this up to f381e4d, as f381e4d seems to solve other problems but is still incomplete? I only tested up to 595e05f thought.

**Comment by 0-8-15** [comment]
I **guess** we can.  Especially as it solves existing problems.

Don't beat me, if it causes problem.

**Comment by mgorges** [comment]
Will hold off - no worries.

**Comment by 0-8-15** [comment]
IMHO these changes could count as drive-by-patches.  I'd apply them. Haven't seen issue here.

**Comment by mgorges** [comment]
I believe this has been superseded by others so can be closed?

**Comment by 0-8-15** [comment]
I believe these are already in master one way or another.  Reopen if there is still any issue.

**Pull request closed without merging**